Sources/Customization/DefaultEventProcessor.swift

```diff
@@ -0,0 +1,271 @@
+/****************************************************************************
+* Copyright 2019, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+
+import Foundation
+
+public enum DataStoreType {
+    case file, memory, userDefaults
+}
+
+open class DefaultEventProcessor: BackgroundingCallbacks, OPTEventProcessor {
+    
+    static let sharedInstance = DefaultEventProcessor()
+    
+    // timer-interval for batching (0 = no batching, negative = use default)
+    var timerInterval: TimeInterval
+    // batch size (1 = no batching, 0 or negative = use default)
+    // attempt to send events in batches with batchSize number of events combined
+    var batchSize: Int
+    var maxQueueSize: Int
+    
+    public struct DefaultValues {
+        static public let batchSize = 10
+        static public let timeInterval: TimeInterval = 60  // secs
```

Comment:
The default batch interval in other SDKs is 30 secs

Reply:
This was discussed before. Default interval should be controlled separately per SDK since it should depend on applications.

Reply:
Ah yeah you are right.